go/tasks/plugins/array/k8s/task.go

```diff
@@ -227,8 +251,34 @@ func (t Task) Finalize(ctx context.Context, tCtx core.TaskExecutionContext, kube
 	indexStr := strconv.Itoa(t.ChildIdx)
 	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
 
+	pod := &v1.Pod{
+		TypeMeta: metaV1.TypeMeta{
+			Kind:       PodKind,
+			APIVersion: v1.SchemeGroupVersion.String(),
+		},
+	}
+
+	err := kubeClient.GetClient().Get(ctx, k8sTypes.NamespacedName{
+		Name:      podName,
+		Namespace: tCtx.TaskExecutionMetadata().GetNamespace(),
```

Comment:
[line 233 above](https://github.com/flyteorg/flyteplugins/pull/186/files#diff-01ee392f44bbd91e1e5f4afd999327e02f8f1df54fd5b5b576478e903cbcb3c2R233) uses a different piece of code to get the namespace... I think we need that.... and given this, maybe we should move the logic that builds this Pod skeleton into a function and just use it in both places?

Reply:
Good catch. This is not building a pod, just name + namespace to query for the pod. There is already a common method for namespace, I changed this to use that here as well.